### DISCUSSION TITLE
Add Support For ValueTask<>

### DISCUSSION THREAD

**Discussion by cportwood** [vkhorikov/CSharpFunctionalExtensions#425] [Ideas]
Currently value tasks are not supported.  I know we can convert any ValueTask<> to a task but it would be nice to have the option to return a ValueTask<>.
Adding support would not be hard. It would just would just be adding a copy of Async versions for each Extension.
Thoughts?

**Comment by cportwood** [comment]
I would be willing to contribute if needed.

**Comment by cportwood** [comment]
Sorry, I did find that this was already suggested.

**Comment by vkhorikov** [comment]
Sorry for late reply, was on vacation. Yes, there already should be a ticket for this, feel free to submit a PR.
